src/module.c

```diff
@@ -5797,7 +5815,11 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
     error_as_call_replies = flags & REDISMODULE_ARGV_CALL_REPLIES_AS_ERRORS;
     va_end(ap);
 
-    c = moduleAllocTempClient();
+    user *user = NULL;
+    if (flags & REDISMODULE_ARGV_CHECK_ACL) {
+        user = ctx->user ? ctx->user->user : ctx->client->user;
+    }
+    c = moduleAllocTempClient(user);
```

Comment:
we "are" using it, but I argue its more consistent to say we only really use it if the "C" argument is given.  Basically I view this set function now as an override for whatever it would think it would be (either not set or something else).

Basically, if you want ACL checking, you need to use "C", if you dont want acl checking you don't use "C".

Reply:
Is `user` used only for ACL checking? I'm not sure...

Reply:
indeed it depends if the user is only used for ACL.
i can see the day where users will have other feature, like rate limiting or whatever.. probably done by a module with some hooks and user awareness.
so it seems logical to me that we always apply the user if it is set.

however, then we run into an odd problem that some ACL checks depend on the "C" flag, and others do not (i.e. if a command does it's own ACL checking, like scripts do, SORT does, and like PUBLISH used to do at some point), these engage even if you didn't pass the "C" flag.
and we can't expect the caller to keep track of our internal implementation..

i'm a bit on the fence here, considering that that built in features of users are just ACL, but it may be extended in the future, and also already be done by modules.

i think i'm leaning towards setting the user regardless of the "C" flag.

Reply:
would we want another flag then?  'U'?  i.e. run as user (where user is either, ctx->client->user or ctx->user (preference being ctx->user).

i.e. could use 'U' without SetContextModuleUser and 'C' and would run scripts not as unrestricted user, but as the appropriate user.  might still be unrestricted.

Reply:
so I thought about this some more.  "C" as I'm arguing to redefine is effectively setting the user for everything.  yeah, it might make sense to not be "C" and be "U" and set user and acl check, but I don't see any case where it makes senes to do one and not do the other.  i.e. if assigning a user to the client, I'd argue you should be acl checking throughout, and as demonstrated for scripts if acl checking at rm_call really should be assigning it to client to impacts script execution

Reply:
so in the commit I just pushed, no "U", "C" is just renamed from ACL_CHECK to RUN_AS_USER.  this does change script semantics that are run via RM_Call so needs to be noted.  ACL checking is just a side effect of "Running as User".